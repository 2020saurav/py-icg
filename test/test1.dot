digraph G 
{

	node0 [label="program"];
	node1 [label= "single_stmt\n[None"];
	node0 -> node1;
	node2 [label= "ENDMARKER\n None]"];
	node0 -> node2;
	node3 [label= "single_stmt\n[None]"];
	node1 -> node3;
	node4 [label= "stmt"];
	node1 -> node4;
	node5 [label= "compound_stmt"];
	node4 -> node5;
	node6 [label= "while_stmt"];
	node5 -> node6;
	node7 [label= "WHILE\n'while'"];
	node6 -> node7;
	node8 [label= "Marker\n{'quad': 0}"];
	node6 -> node8;
	node9 [label= "test"];
	node6 -> node9;
	node10 [label= "COLON\n':'"];
	node6 -> node10;
	node11 [label= "Marker\n{'quad': 0}"];
	node6 -> node11;
	node12 [label= "suite"];
	node6 -> node12;
	node13 [label= "NEWLINE\n'\\n'"];
	node12 -> node13;
	node14 [label= "INDENT"];
	node12 -> node14;
	node15 [label= "stmts"];
	node12 -> node15;
	node16 [label= "DEDENT"];
	node12 -> node16;
	node17 [label= "stmt"];
	node15 -> node17;
	node18 [label= "simple_stmt"];
	node17 -> node18;
	node19 [label= "small_stmts"];
	node18 -> node19;
	node20 [label= "NEWLINE\n'\\n'"];
	node18 -> node20;
	node21 [label= "small_stmt"];
	node19 -> node21;
	node22 [label= "expr_stmt"];
	node21 -> node22;
	node23 [label= "test"];
	node22 -> node23;
	node24 [label= "EQUAL\n'='"];
	node22 -> node24;
	node25 [label= "test"];
	node22 -> node25;
	node26 [label= "or_test"];
	node25 -> node26;
	node27 [label= "and_test"];
	node26 -> node27;
	node28 [label= "ortestlist"];
	node26 -> node28;
	node29 [label= "not_test"];
	node27 -> node29;
	node30 [label= "andtestlist"];
	node27 -> node30;
	node31 [label= "comparison"];
	node29 -> node31;
	node32 [label= "expr"];
	node31 -> node32;
	node33 [label= "compexprlist"];
	node31 -> node33;
	node34 [label= "xor_expr"];
	node32 -> node34;
	node35 [label= "xorexprlist"];
	node32 -> node35;
	node36 [label= "and_expr"];
	node34 -> node36;
	node37 [label= "andexprlist"];
	node34 -> node37;
	node38 [label= "shift_expr"];
	node36 -> node38;
	node39 [label= "shiftexprlist"];
	node36 -> node39;
	node40 [label= "arith_expr"];
	node38 -> node40;
	node41 [label= "arithexprlist"];
	node38 -> node41;
	node42 [label= "term"];
	node40 -> node42;
	node43 [label= "termlist"];
	node40 -> node43;
	node44 [label= "MINUS\n'-'"];
	node43 -> node44;
	node45 [label= "term"];
	node43 -> node45;
	node46 [label= "termlist"];
	node43 -> node46;
	node47 [label= "factor"];
	node45 -> node47;
	node48 [label= "factorlist"];
	node45 -> node48;
	node49 [label= "power"];
	node47 -> node49;
	node50 [label= "atom"];
	node49 -> node50;
	node51 [label= "trailerlist"];
	node49 -> node51;
	node52 [label= "NUMBER\n1"];
	node50 -> node52;
	node53 [label= "factor"];
	node42 -> node53;
	node54 [label= "factorlist"];
	node42 -> node54;
	node55 [label= "power"];
	node53 -> node55;
	node56 [label= "atom"];
	node55 -> node56;
	node57 [label= "trailerlist"];
	node55 -> node57;
	node58 [label= "NAME\n'i'"];
	node56 -> node58;
	node59 [label= "or_test"];
	node23 -> node59;
	node60 [label= "and_test"];
	node59 -> node60;
	node61 [label= "ortestlist"];
	node59 -> node61;
	node62 [label= "not_test"];
	node60 -> node62;
	node63 [label= "andtestlist"];
	node60 -> node63;
	node64 [label= "comparison"];
	node62 -> node64;
	node65 [label= "expr"];
	node64 -> node65;
	node66 [label= "compexprlist"];
	node64 -> node66;
	node67 [label= "xor_expr"];
	node65 -> node67;
	node68 [label= "xorexprlist"];
	node65 -> node68;
	node69 [label= "and_expr"];
	node67 -> node69;
	node70 [label= "andexprlist"];
	node67 -> node70;
	node71 [label= "shift_expr"];
	node69 -> node71;
	node72 [label= "shiftexprlist"];
	node69 -> node72;
	node73 [label= "arith_expr"];
	node71 -> node73;
	node74 [label= "arithexprlist"];
	node71 -> node74;
	node75 [label= "term"];
	node73 -> node75;
	node76 [label= "termlist"];
	node73 -> node76;
	node77 [label= "factor"];
	node75 -> node77;
	node78 [label= "factorlist"];
	node75 -> node78;
	node79 [label= "power"];
	node77 -> node79;
	node80 [label= "atom"];
	node79 -> node80;
	node81 [label= "trailerlist"];
	node79 -> node81;
	node82 [label= "NAME\n'i'"];
	node80 -> node82;
	node83 [label= "or_test"];
	node9 -> node83;
	node84 [label= "and_test"];
	node83 -> node84;
	node85 [label= "ortestlist"];
	node83 -> node85;
	node86 [label= "not_test"];
	node84 -> node86;
	node87 [label= "andtestlist"];
	node84 -> node87;
	node88 [label= "comparison"];
	node86 -> node88;
	node89 [label= "expr"];
	node88 -> node89;
	node90 [label= "compexprlist"];
	node88 -> node90;
	node91 [label= "comp_op"];
	node90 -> node91;
	node92 [label= "expr"];
	node90 -> node92;
	node93 [label= "compexprlist"];
	node90 -> node93;
	node94 [label= "xor_expr"];
	node92 -> node94;
	node95 [label= "xorexprlist"];
	node92 -> node95;
	node96 [label= "and_expr"];
	node94 -> node96;
	node97 [label= "andexprlist"];
	node94 -> node97;
	node98 [label= "shift_expr"];
	node96 -> node98;
	node99 [label= "shiftexprlist"];
	node96 -> node99;
	node100 [label= "arith_expr"];
	node98 -> node100;
	node101 [label= "arithexprlist"];
	node98 -> node101;
	node102 [label= "term"];
	node100 -> node102;
	node103 [label= "termlist"];
	node100 -> node103;
	node104 [label= "factor"];
	node102 -> node104;
	node105 [label= "factorlist"];
	node102 -> node105;
	node106 [label= "power"];
	node104 -> node106;
	node107 [label= "atom"];
	node106 -> node107;
	node108 [label= "trailerlist"];
	node106 -> node108;
	node109 [label= "NUMBER\n0"];
	node107 -> node109;
	node110 [label= "GREATER\n'>'"];
	node91 -> node110;
	node111 [label= "xor_expr"];
	node89 -> node111;
	node112 [label= "xorexprlist"];
	node89 -> node112;
	node113 [label= "and_expr"];
	node111 -> node113;
	node114 [label= "andexprlist"];
	node111 -> node114;
	node115 [label= "shift_expr"];
	node113 -> node115;
	node116 [label= "shiftexprlist"];
	node113 -> node116;
	node117 [label= "arith_expr"];
	node115 -> node117;
	node118 [label= "arithexprlist"];
	node115 -> node118;
	node119 [label= "term"];
	node117 -> node119;
	node120 [label= "termlist"];
	node117 -> node120;
	node121 [label= "factor"];
	node119 -> node121;
	node122 [label= "factorlist"];
	node119 -> node122;
	node123 [label= "power"];
	node121 -> node123;
	node124 [label= "atom"];
	node123 -> node124;
	node125 [label= "trailerlist"];
	node123 -> node125;
	node126 [label= "NAME\n'i'"];
	node124 -> node126;
	node127 [label= "single_stmt\n[]"];
	node3 -> node127;
	node128 [label= "stmt"];
	node3 -> node128;
	node129 [label= "simple_stmt"];
	node128 -> node129;
	node130 [label= "small_stmts"];
	node129 -> node130;
	node131 [label= "NEWLINE\n'\\n'"];
	node129 -> node131;
	node132 [label= "small_stmt"];
	node130 -> node132;
	node133 [label= "expr_stmt"];
	node132 -> node133;
	node134 [label= "test"];
	node133 -> node134;
	node135 [label= "EQUAL\n'='"];
	node133 -> node135;
	node136 [label= "test"];
	node133 -> node136;
	node137 [label= "or_test"];
	node136 -> node137;
	node138 [label= "and_test"];
	node137 -> node138;
	node139 [label= "ortestlist"];
	node137 -> node139;
	node140 [label= "not_test"];
	node138 -> node140;
	node141 [label= "andtestlist"];
	node138 -> node141;
	node142 [label= "comparison"];
	node140 -> node142;
	node143 [label= "expr"];
	node142 -> node143;
	node144 [label= "compexprlist"];
	node142 -> node144;
	node145 [label= "xor_expr"];
	node143 -> node145;
	node146 [label= "xorexprlist"];
	node143 -> node146;
	node147 [label= "and_expr"];
	node145 -> node147;
	node148 [label= "andexprlist"];
	node145 -> node148;
	node149 [label= "shift_expr"];
	node147 -> node149;
	node150 [label= "shiftexprlist"];
	node147 -> node150;
	node151 [label= "arith_expr"];
	node149 -> node151;
	node152 [label= "arithexprlist"];
	node149 -> node152;
	node153 [label= "term"];
	node151 -> node153;
	node154 [label= "termlist"];
	node151 -> node154;
	node155 [label= "factor"];
	node153 -> node155;
	node156 [label= "factorlist"];
	node153 -> node156;
	node157 [label= "power"];
	node155 -> node157;
	node158 [label= "atom"];
	node157 -> node158;
	node159 [label= "trailerlist"];
	node157 -> node159;
	node160 [label= "NUMBER\n10"];
	node158 -> node160;
	node161 [label= "or_test"];
	node134 -> node161;
	node162 [label= "and_test"];
	node161 -> node162;
	node163 [label= "ortestlist"];
	node161 -> node163;
	node164 [label= "not_test"];
	node162 -> node164;
	node165 [label= "andtestlist"];
	node162 -> node165;
	node166 [label= "comparison"];
	node164 -> node166;
	node167 [label= "expr"];
	node166 -> node167;
	node168 [label= "compexprlist"];
	node166 -> node168;
	node169 [label= "xor_expr"];
	node167 -> node169;
	node170 [label= "xorexprlist"];
	node167 -> node170;
	node171 [label= "and_expr"];
	node169 -> node171;
	node172 [label= "andexprlist"];
	node169 -> node172;
	node173 [label= "shift_expr"];
	node171 -> node173;
	node174 [label= "shiftexprlist"];
	node171 -> node174;
	node175 [label= "arith_expr"];
	node173 -> node175;
	node176 [label= "arithexprlist"];
	node173 -> node176;
	node177 [label= "term"];
	node175 -> node177;
	node178 [label= "termlist"];
	node175 -> node178;
	node179 [label= "factor"];
	node177 -> node179;
	node180 [label= "factorlist"];
	node177 -> node180;
	node181 [label= "power"];
	node179 -> node181;
	node182 [label= "atom"];
	node181 -> node182;
	node183 [label= "trailerlist"];
	node181 -> node183;
	node184 [label= "NAME\n'i'"];
	node182 -> node184;
}
