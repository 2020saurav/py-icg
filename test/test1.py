for i in [1,2,3,4]:
	if i == 2:
		continue
	else:
		i = 3