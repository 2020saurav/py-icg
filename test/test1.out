program :
    0: 	['var1', 5, '', '=']					✓ var2 is index
    1: 	['var2', 0, '', '=']					✓
    2: 	['var3', [1, 2, 3], 'ARRAY', '=']		✓
    3: 	['var4', 3, '', '=']					✓ Len
    4: 	['var5', 'var2', 3, '==']				✓ Contidition
    5: 	['var5', 1, 34, 'COND_GOTO']			✓ Jump to outside for loop
    6: 	['var1', 'var3[var2]', '', '=']			✓ set var 1
    7: 	['var6', 5, '', '=']					✓
    8: 	['var7', 'var6', 10, '<']				✓
    9: 	['var7', 0, 27, 'COND_GOTO']			✓
   10: 	['var8', 'var6', 8, '==']				✓
   11: 	['var8', 0, 15, 'COND_GOTO']			✓
   12: 	['var6', '', 'NUMBER', 'PRINT']			✓
   13: 	['"\n"', '', 'STRING', 'PRINT']			✓
   14: 	['', '', 24, 'GOTO']					✓
   15: 	['var9', 'var6', 5, '>']				✓
   16: 	['var9', 0, 22, 'COND_GOTO']			✓
   17: 	['var10', 'var6', 'var1', '*']			✓
   18: 	['var10', '', 'NUMBER', 'PRINT']		✓
   19: 	['"\n"', '', 'STRING', 'PRINT']			✓
   20: 	['', '', 27, 'GOTO']					✓
   21: 	['', '', 24, 'GOTO']					✓
   22: 	['"HI"', '', 'STRING', 'PRINT']			✓
   23: 	['"\n"', '', 'STRING', 'PRINT']			✓
   24: 	['var11', 'var6', 'var1', '+']			✓
   25: 	['var6', 'var11', '', '=']				✓
   26: 	['', '', 8, 'GOTO']						✓
   27: 	['var12', 'var1', 2, '==']				✓
   28: 	['var12', 0, 31, 'COND_GOTO']			✓
   29: 	['', '', -1, 'GOTO']					X
   30: 	['', '', 32, 'GOTO']					✓
   31: 	['var1', 3, '', '=']					✓
   32: 	['var2', 'var2', 1, '+']				✓
   33: 	['', '', 4, 'GOTO']						✓
   34: 	['"hello"', '', 'STRING', 'PRINT']		✓
   35: 	['"\n"', '', 'STRING', 'PRINT']			✓
   36: 	['', '', -1, 'HALT']					✓